tests/datastore/test_async_sqlite_datastore.nim

```diff
@@ -0,0 +1,142 @@
+import std/options
+import std/os
+
+import asynctest/unittest2
+import pkg/stew/results
+
+import ../../datastore/async_sqlite_datastore
+
+const
+  bytes = @[1.byte, 2.byte, 3.byte]
+  key = Key.init("a:b/c/d:e").get
+
+var
+  ds {.threadvar.}: AsyncSQLiteDatastore
+  basePath {.threadvar.}: string
+  basePathAbs {.threadvar.}: string
+  filename {.threadvar.}: string
+  dbPathAbs {.threadvar.}: string
```

Comment:
When using `asynctest/unittest2` I wasn't able to include these in the `setup:` block like I can when using `unittest2`, I'm not sure why.

Also, to avoid a compiler error re: gc safety, I "cheated" by marking them as threadvars; seemed like an okay workaround for the purposes of a test suite.

Reply:
asynctest puts the whole of body of `suite` into a proc, so you don't need the `{.threadvar.}` because the variables won't be global.

Reply:
if you want isolation between tests, variables should generally go inside a `setup`, else the instance is reused between tests - websock tests suffered from this, where an instance created in one test was used for the next test - it's generally good practice to not have dependencies like this - see https://github.com/status-im/nim-websock/pull/115

Reply:
also, most nim projects we have prefer the name `asyncTest` for async tests so as to highlight the additional failure modes of running with potential execution reordering

Reply:
@arnetheduck from where to import asyncTest ? 

I find one in https://github.com/status-im/nim-chronos/blob/master/chronos/unittest2/asynctests.nim
- is it proper place?